### DISCUSSION TITLE
updateListingInventory  - Update only Quantity in  Variations

### DISCUSSION THREAD

**Discussion by pichutan** [etsy/open-api#1176] [Q&A] [answered]
Hello there,

I am playing around with insomnia to get used to the api. Currently I am trying to update the quantity of an offering. 

My request for products looks like this: `{"sku": "XXX", "offerings": [{"quantity":14, "price": 10, "is_enabled": true}]}`. The quantity got updated but with that all of the existing variations got deleted as well. 

How can I update the quantity of a product variation - without deleting the other variations?

Does Etsy charge 0,20 cents for every update then?

Cheers

**Comment by dmayes403** [comment]
@pichutan I'm looking into doing something similar. Which API are you using to do this?

**Comment by showtroylove** [comment, marked as answer]
You cannot updated a single ListingInventory record you must perform a POST operation and include the entire dataset of your inventory.

**Reply by mxmbt2** [reply]
By "entire dataset" do you just mean all variations for the listing, right? Not the entire dataset for the shop...

**Reply by showtroylove** [reply]
Correct. You cannot partially edit variation on the listing.  You must supply all each time you make changes.

**Reply by mxmbt2** [reply]
Thanks for replying. I'm building a python script to update qty and price for a single listing. First I download all the variation data from the listing merge that data with the qty and price data change then put the file to Etsy. But I'm struggling with the final step where I'm just getting errors like:

Failed to update the listing variations: 400, {"error":"Array contains invalid keys: product_id,is_deleted"}

Is there anywhere that has a good example of the appropriate JSON package and posting script for variation updates? Or a tutorial? Any assistance would be deeply appreciated. Thanks in advance.

**Reply by showtroylove** [reply]
Call the getListingInventory method on an existing listing to get an example of the json payload.   Ensure your json follows the pattern precisely.  It is very sensitive.  So I recommend use postman or similar to test small tweaks to the json and submit it to Etsy for feedback.

Ensure you're not including properties in your json (i.e. is_deleted) because it strictly interprets the json.  Meaning only explicitly required and optional properties defined in the API Reference are permitted.

**Reply by TuanNQT** [reply]
> Call the getListingInventory method on an existing listing to get an example of the json payload. Ensure your json follows the pattern precisely. It is very sensitive. So I recommend use postman or similar to test small tweaks to the json and submit it to Etsy for feedback.
> 
> Ensure you're not including properties in your json (i.e. is_deleted) because it strictly interprets the json. Meaning only explicitly required and optional properties defined in the API Reference are permitted.

I got "error ":"Server Error" 
![image](https://github.com/user-attachments/assets/a04e4089-4678-45ff-9815-abe1388dc757)
